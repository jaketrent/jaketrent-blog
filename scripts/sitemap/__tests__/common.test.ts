import {
  fileFromRoot,
  formatUrl,
  formatUrlset,
  parseUrlFromContentPath,
} from "../common"

describe("#fileFromRoot", () => {
  it("converts concrete paths", () => {
    expect(fileFromRoot("public/static.xml")).toEqual("public/static.xml")
  })
  it("converts glob paths", () => {
    expect(fileFromRoot("pages/**/*.tsx")).toEqual("pages/**/*.tsx")
  })
})

describe("#parseUrlFromContentPath", () => {
  it("removes dir path and extension", () => {
    expect(parseUrlFromContentPath("content/post/page.md")).toEqual(
      "/post/page"
    )
  })
})

describe("#formatUrl", () => {
  it("renders", () => {
    expect(formatUrl("mockISODateString", "/post/test-post")).toMatchSnapshot()
  })
})

describe("#formatUrlset", () => {
  it("renders", () => {
    expect(formatUrlset("<mockFormattedUrls />")).toMatchSnapshot()
  })
})
