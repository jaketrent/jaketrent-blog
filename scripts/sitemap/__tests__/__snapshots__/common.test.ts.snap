// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#formatUrl renders 1`] = `
"
  <url>
    <loc>https://jaketrent.com/post/test-post</loc>
    <lastmod>mockISODateString</lastmod>
  </url>
"
`;

exports[`#formatUrlset renders 1`] = `
"
  <?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
  <urlset
    xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\"
    xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\"
    xsi:schemaLocation=\\"http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd\\"
  >
    <mockFormattedUrls />
  </urlset>
"
`;
